core/import.go

```diff
@@ -17,32 +17,50 @@ import (
 	"time"
 )
 
+const bufferSize = 5
+
 func (n *OpenBazaarNode) ImportListings(r io.ReadCloser) error {
 	reader := csv.NewReader(r)
-	records, err := reader.ReadAll()
+	columns, err := reader.Read()
 	if err != nil {
 		return err
 	}
-	if len(records) < 2 {
-		return errors.New("Invalid csv file")
-	}
-	columns := records[0]
 	fields := make(map[string]int)
 	for i, c := range columns {
 		fields[strings.ToLower(c)] = i
 	}
 
-	type Result struct {
-		listing *pb.Listing
-		err     error
-	}
+	done := make(chan struct{})
+	buf := make(chan struct{}, bufferSize)
+	errChan := make(chan error)
+
+	countLock := new(sync.Mutex)
+	count := 0
 
-	respCh := make(chan Result, len(records)-1)
-	defer close(respCh)
+	indexLock := new(sync.Mutex)
+
+	for {
+		select {
+		case err := <-errChan:
+			return err
+		case <-done:
```

Comment:
If the last 5 goroutines are in flight and one finishes, next time through the loop it will try to read from the CSV and get an end of file error then `done` will be called. I suppose it's possible `done` can be called before the remaining four goroutines finish.

How can we make sure the remaining four goroutines finish so we can check for errors before returning.?